src/shell/shell.cpp

```diff
@@ -736,6 +754,98 @@ class AUTOEXEC final : public Module_base {
 	}
 };
 
+// Specify a 'Drive' config object with allowed key and value types
+static Config specify_drive_config()
+{
+	auto conf = Config();
+
+	// Define the [drive] section
+	constexpr auto changeable_at_runtime = false;
+	auto prop = conf.AddSection_prop("drive", nullptr, changeable_at_runtime);
+
+	// Define the allowed keys and types
+	constexpr auto on_startup = Property::Changeable::OnlyAtStart;
+	const char *drive_types[] = {"dir", "floppy", "cdrom", "overlay", nullptr};
+	(void)prop->Add_string("type", on_startup, "")->Set_values(drive_types);
+	(void)prop->Add_string("label", on_startup, "");
+	(void)prop->Add_string("path", on_startup, "");
+	(void)prop->Add_string("override_drive", on_startup, "");
+	(void)prop->Add_bool("verbose", on_startup, true);
+
+	return conf;
+}
+
+// Parse a 'Drive' config file and return object with allowed key and value types
+static std::tuple<std::string, std::string, std::string> parse_drive_conf(
+        std::string drive_letter, const std_fs::path &conf_path)
```

Comment:
const std::string & ?

Reply:
I'm using it as a local copy, and do write to it:

``` c++
if (override_drive.length() == 1 && override_drive[0] >= 'a' && override_drive[0] <= 'y')
	drive_letter = override_drive;
```

(Good catch; I'm really liking our "read-only-arguments unless we want to write to them inside the function" approach: it's self-document how we use them, and you read it that way)